#168781629 - Users Filter Requests

nignanthomas opened the pull request:
#### What does this PR do?
Users Filter Requests.

#### Description of Task to be completed?
- Create searchBar Component with fields
- Create actions to fetch filtered requests

#### How should this be manually tested?
- Clone ft-filter-requests-168781629 branch
- run `npm i`
- run `npm run dev`
- Click on `Requests` in the MenuBar
- Click on `Filter Requests` to show/hide the Search Bar

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#168781629](https://www.pivotaltracker.com/story/show/168781629)

#### Screenshots (if appropriate)
<img width="1440" alt="Screen Shot 2019-12-19 at 18 03 09" src="https://user-images.githubusercontent.com/42959540/71189055-e02b8080-228a-11ea-96da-8978c56b2861.png">
<img width="1439" alt="Screen Shot 2019-12-19 at 18 03 22" src="https://user-images.githubusercontent.com/42959540/71189056-e0c41700-228a-11ea-9fa6-d372bf8c708d.png">
<img width="1428" alt="Screen Shot 2019-12-19 at 18 09 44" src="https://user-images.githubusercontent.com/42959540/71189059-e0c41700-228a-11ea-95ec-083fe0d818d2.png">

#### Questions:
N/A

nignanthomas committed:
feature(filter-requests): Users Filter Requests

- create searchBar Component with fields

- create actions to fetch filtered requests

[Finishes #168781629]

Cheza-Dzabala reviewed (commented):
@nignanthomas, looks good.. However, a user should not be able to enter a request id from the front end

alainmateso commented on src/components/pages/requests/SearchComponent.js:
```diff
@@ -0,0 +1,168 @@
+import { searchRequestAction, getRequestsAction } from '../../../actions/requestsActions';
+import Breadcrumbs from '../../global/Breadcrumbs';
+// import RequestItem from './RequestItem';
+
```
remove commented code

alainmateso commented on src/components/pages/requests/SearchComponent.js:
```diff
@@ -0,0 +1,168 @@
+    if (searchParams !== '') {
+      searchParams = `?${searchParams}`;
+    }
+    console.log('searchParams ===> ', searchParams);
```
remove console.log statements

alainmateso commented on src/components/pages/requests/SearchComponent.js:
```diff
@@ -0,0 +1,168 @@
+            <Col xs={12} sm={6} md={4} lg={4}>
+              <Form.Group>
+                <Form.Control as="input" type="date" name="departureDate" onChange={this.handleChange} />
+                {/* <DayPickerInput /> */}
```
remove commented code

alainmateso commented on src/views/requests/ViewRequests.js:
```diff
@@ -25,17 +26,49 @@ export class ViewRequests extends Component {
+    const { status, searchData, searchDataError } = searchRequests;
+    console.log('searchStatus ===> ', status);
+    console.log('searchData ===> ', searchData);
+    console.log('searchDataError ===> ', searchDataError);
```
remove console.log statements

alainmateso commented on src/views/requests/ViewRequests.js:
```diff
@@ -25,17 +26,49 @@ export class ViewRequests extends Component {
+    console.log('searchData ===> ', searchData);
+    console.log('searchDataError ===> ', searchDataError);
+
+    console.log('searchRequests ===> ', searchRequests);
```
remove console.log statements

alainmateso commented on src/views/requests/ViewRequests.js:
```diff
@@ -25,17 +26,49 @@ export class ViewRequests extends Component {
+    // if (status !== '') {
+    //   data = props.searchData;
+    //   dataError = props.searchDataError;
+    // }
```
remove commented code

alainmateso commented on src/views/requests/ViewRequests.js:
```diff
@@ -57,19 +90,28 @@ export class ViewRequests extends Component {
           </Row>
         ))}
         <Row>
           {dataError && <Alert variant="danger" heading="Error" message={dataError.message} />}
```
remove commented code

alainmateso commented on src/views/requests/ViewRequests.js:
```diff
@@ -78,6 +120,13 @@ export class ViewRequests extends Component {
+  // searchData: state.searchRequests.data,
+  // searchDataError: state.searchRequests.dataError,
+  // searchStatus: state.searchRequests.status,
+
```
remove commented code

alainmateso reviewed (changes requested):
Great job @nignanthomas . Kindly work on the following feedback
- Since the id of the request is not displayed anywhere on the request information, kindly add it so that the search by id makes sense. 
- When no results are found, I believe it is better to use any other selector like just a label to tell the user that no requests were found instead of bring the alert component

I've also left a few comments, please check them out. Impressive work 👍🏾

nakiwuge commented on src/components/pages/requests/SearchBar.js:
```diff
@@ -0,0 +1,171 @@
+  searchRequest = async () => {
+    const { state, props } = this;
+    let searchParams = '';
+    const { isLoading, openSearch, ...options } = state;
```
Add a new line

nakiwuge commented on src/components/pages/requests/SearchBar.js:
```diff
@@ -0,0 +1,171 @@
+        searchParams += `${key}=${options[`${key}`]}&`;
+      }
+    });
+    searchParams = searchParams.substring(0, searchParams.length - 1);
```
Add a new line

nakiwuge commented on src/components/pages/requests/SearchBar.js:
```diff
@@ -0,0 +1,171 @@
+      if (options[`${key}`] !== '') {
+        searchParams += `${key}=${options[`${key}`]}&`;
+      }
+    });
```
Add a new line

nakiwuge commented on src/components/pages/requests/SearchBar.js:
```diff
@@ -0,0 +1,171 @@
+    searchParams = searchParams.substring(0, searchParams.length - 1);
+    if (searchParams !== '') {
+      searchParams = `?${searchParams}`;
+    }
```
Add a new line

nakiwuge commented on src/components/pages/requests/SearchBar.js:
```diff
@@ -0,0 +1,171 @@
+  render() {
+    const { props, state } = this;
+    const { isLoading, openSearch } = state;
+    const { locations } = props;
```
Add a new line

nakiwuge commented on src/components/pages/requests/SearchBar.js:
```diff
@@ -0,0 +1,171 @@
+      { openSearch && (
+        <Container className="search-bar">
+          <Row className="center-items">
+            {/* <Col xs={12} sm={6} md={4} lg={4}>
```
Remove commented code

nakiwuge commented on src/styles/requests/requests.scss:
```diff
@@ -1,5 +1,5 @@
 .request-item {
-  background-color: $light;
+  background-color: #ffffff;
```
move this color to the variables

nakiwuge commented on src/styles/breadcrumbs.scss:
```diff
@@ -1,3 +1,6 @@
 	width: 40%;
 }
+.bc-bg {
+	background-color: #ffffff !important;
```
move this color to the variables

nakiwuge reviewed (changes requested):
@nignanthomas this is great work however you have a date picker as one of the options but it doesn't have a label so a user cannot know whether its a date for creating a request or arrival date or destination date.

I have left other comments on the pr kindly check them out

nignanthomas committed:
feature(filter-requests): Users Filter Requests

    - create searchBar Component with fields

    - create actions to fetch filtered requests

    [Finishes #168781629]
